### PR TITLE
Fix the issue that PrerollContext.has_texture_layer is sticky.

### PR DESCRIPTION
**Pull request by guoguo338**
Fix the issue for PrerollContext.has_texture_layer is sticky and impacts layers unrelated to the TextureLayer, link as below:
https://github.com/flutter/flutter/issues/103013

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: flow/layers/container_layer_unittests.cc [hunk 1]
```diff
@@ -55,6 +63,34 @@ TEST_F(ContainerLayerTest, PaintBeforePrerollDies) {
...
+  SkPaint child_paint2(SkColors::kGreen);
+
+  auto mock_layer1 = std::make_shared<MockLayer>(
+      child_path1, child_paint1, false /* fake_has_platform_view */, false,
+      false, true /* fake_has_texture_layer */);
+  auto mock_layer2 = std::make_shared<MockLayer>(child_path2, child_paint2);
+
+  auto root = std::make_shared<ContainerLayer>();
+  auto container_layer1 = std::make_shared<ContainerLayer>();
+  auto container_layer2 = std::make_shared<ContainerLayer>();
```
Comment (flar) [inline flow/layers/container_layer_unittests.cc:81]: container_layer2 is added to root, but it is never given any children. Was mock_2 supposed to be placed in container_2? Either way, the test is valid, but the existence of container_2 looks odd.

File: flow/testing/mock_layer.h [hunk 2]
```diff
@@ -28,7 +28,8 @@ class MockLayer : public Layer {
                      SkPaint paint = SkPaint(),
                      bool fake_has_platform_view = false,
                      bool fake_reads_surface = false,
-                     bool fake_opacity_compatible_ = false);
+                     bool fake_opacity_compatible_ = false,
+                     bool fake_has_texture_layer = false);
```
[the remark this answers is not in the document]
Comment (flar) [reply flow/testing/mock_layer.h:32]: @ColdPaleLight - thoughts...? Is this becoming a maintenance trap?

File: flow/layers/container_layer_unittests.cc (hunk 1 above)
Comment (guoguo338) [reply flow/layers/container_layer_unittests.cc:81]: done!

**Review by ColdPaleLight — APPROVED** [review]
LGTM!

File: flow/testing/mock_layer.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (ColdPaleLight) [reply flow/testing/mock_layer.h:32]: Yes, bit flags sound great. We will file a new PR to address it.

**Pull request merged**